cluster/up.sh

```diff
@@ -4,54 +4,8 @@ source ./cluster/gocli.sh
 source ./hack/build/config.sh
 source ./cluster/$KUBEVIRT_PROVIDER/provider.sh
 
-num_nodes=${KUBEVIRT_NUM_NODES:-1}
-mem_size=${KUBEVIRT_MEMORY_SIZE:-5120M}
-
-re='^-?[0-9]+$'
-if ! [[ $num_nodes =~ $re ]] || [[ $num_nodes -lt 1 ]] ; then
-    num_nodes=1
-fi
-
-image=$(getClusterType)
 echo "Image:${image}"
-if [[ $KUBEVIRT_PROVIDER == "external" ]]; then
-  echo "using external cluster"
-elif [[ $image == $KUBERNETES_IMAGE ]]; then
-    $gocli run --enable-ceph --random-ports --nodes ${num_nodes} --memory ${mem_size} --background kubevirtci/${image}
-    cluster_port=$($gocli ports k8s | tr -d '\r')
-    $gocli scp /usr/bin/kubectl - > ./cluster/.kubectl
-    chmod u+x ./cluster/.kubectl
-    $gocli scp /etc/kubernetes/admin.conf - > ./cluster/.kubeconfig
-    export KUBECONFIG=./cluster/.kubeconfig
-    ./cluster/.kubectl config set-cluster kubernetes --server=https://127.0.0.1:$cluster_port
-    ./cluster/.kubectl config set-cluster kubernetes --insecure-skip-tls-verify=true
-
-elif [[ $image == $OPENSHIFT_IMAGE ]]; then
-
-    # If on a developer setup, expose ocp on 8443, so that the openshift web console can be used (the port is important because of auth redirects)
-    if [ -z "${JOB_NAME}" ]; then
-        KUBEVIRT_PROVIDER_EXTRA_ARGS="${KUBEVIRT_PROVIDER_EXTRA_ARGS} --ocp-port 8443"
-    fi
-
-    $gocli run --random-ports --reverse --nodes ${num_nodes} --memory ${mem_size} --background kubevirtci/${image} ${KUBEVIRT_PROVIDER_EXTRA_ARGS}
-    cluster_port=$($gocli ports ocp | tr -d '\r')
-    $gocli scp /etc/origin/master/admin.kubeconfig - > ./cluster/.kubeconfig
-    $gocli ssh node01 -- sudo cp /etc/origin/master/admin.kubeconfig ~vagrant/
-    $gocli ssh node01 -- sudo chown vagrant:vagrant ~vagrant/admin.kubeconfig
-
-    # Copy oc tool and configuration file
-    $gocli scp /usr/bin/oc - >./cluster/.kubectl
-    chmod u+x ./cluster/.kubectl
-    $gocli scp /etc/origin/master/admin.kubeconfig - > ./cluster/.kubeconfig
-    # Update Kube config to support unsecured connection
-    export KUBECONFIG=./cluster/.kubeconfig
-    ./cluster/.kubectl config set-cluster node01:8443 --server=https://127.0.0.1:$cluster_port
-    ./cluster/.kubectl config set-cluster node01:8443 --insecure-skip-tls-verify=true
-fi
-
-echo 'Wait until all nodes are ready'
-until [[ $(_kubectl get nodes --no-headers | wc -l) -eq $(_kubectl get nodes --no-headers | grep " Ready" | wc -l) ]]; do
-    sleep 1
-done
-echo "cluster node are ready!"
+echo "Starting cluster"
+up
 
+verify
```

Comment:
If you want to add this like this; then you need to also make sure you add it to the external/provider.sh file.  Going forward we'd also need to add it to local if/when we add that.

Reply:
I don't since I am sourcing the ephemeral-provider.sh which defines the function. So if a specific provider doesn't care about it, and wants to use the one from the ephemeral-provider.sh it doesn't have to add it.

Reply:
Sure, but that breaks the external provider; because the externa/provider.sh file does not have a verify function; and of course you can't use the ephemeral provider scripts for the external provider.

Just add a verify function to the external provider and you're good.

Reply:
All you need to do is add a function to the external/provider.sh file:

```
function verify(){
  echo "no verification performed on external providers"
}
```

Reply:
One other nit on this PR... so the title is "Updated k8s image to latest"; yes that's true, but at the same time there's a refactor of the provider scripts for deploying clusters.  That's ok; but I think you're changing enough here (inheritance model closer to kubevirts) that it warrants its own pr.  Not a big deal, but going forward I think it's worth considering.

Reply:
I changed the title of the PR to better convey what it is doing, not the commit message is more descriptive.

Reply:
I thought I was sourcing the ephemeral provider in the external one, and I wasn't so yes I am missing the verify function, I will add.